module-bastion.tf

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2017, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+locals {
+  bastion_public_ip = (var.create_bastion
+    ? one(module.bastion[*].public_ip)
+    : var.bastion_public_ip
+  )
+
+  bastion_image_id = (var.bastion_image_type == "custom"
+    ? var.bastion_image_id : element(tolist(setintersection([
+      lookup(local.image_ids, format("%s %s", var.bastion_image_os, split(".", var.bastion_image_os_version)[0]), null),
+      local.image_ids.nongpu, local.image_ids.x86_64,
+    ]...)), 0)
+  )
+}
+
+module "bastion" {
+  count          = var.create_bastion ? 1 : 0
+  source         = "./modules/bastion"
```

Comment:
Remote submodules such as vcn, bastion and operator are reusable modules and not just for this project.

If there are enhancements you wish to bring to them, then they should equally apply for the remote and should be implemented there first instead of embedding this locally. We moved away from embedding locally so we don't have to maintain the code in 2 different places.

Reply:
Discussed and in agreement for now - will revisit in a subsequent PR.